core/trino-main/src/main/java/io/trino/connector/WorkerDynamicCatalogManager.java

```diff
@@ -45,8 +50,9 @@ public class WorkerDynamicCatalogManager
 
     private final CatalogFactory catalogFactory;
 
-    private final Lock catalogsUpdateLock = new ReentrantLock();
+    private final ReadWriteLock catalogsUpdateLock = new ReentrantReadWriteLock();
     private final ConcurrentMap<CatalogHandle, CatalogConnector> catalogs = new ConcurrentHashMap<>();
+    private final ExecutorService catalogWorkService = Executors.newVirtualThreadPerTaskExecutor();
```

Comment:
"service" is a misnomer here. Rename it to `executor` (no need to mention "catalogs", since this is the sole purpose of `WorkerDynamicCatalogManager`

Reply:
👍